### ISSUE TITLE
[Feature Request] Workflow and Dialog Systems

### ISSUE THREAD

**Issue by AdamSobieski** [microsoft/BotBuilder-Samples#3766]
I would like to request a Bot Builder sample which showcases coordination and orchestration between workflow engines (e.g., [Azure Logic Apps](https://docs.microsoft.com/en-us/azure/logic-apps/) and [others](https://github.com/meirwah/awesome-workflow-engines)) and dialog systems.

A use case of interest is dialog systems which are capable of engaging in dialogs about activities, e.g., mathematics homework exercises, which are selected and presented in sequences to end-users during sessions, e.g., mathematics homework sessions.

A simple workflow activity derived from a mathematics exercise might present an end-user with a mathematics question, present them with multiple choices, process a timer, and await a response. Interestingly, workflow-based approaches are recursive (activities can contain sub-activities) and scalable and extensible to more complex scenarios.

Dialog systems can be envisioned which simultaneously utilize inter-session, session-based, and per-activity context data as they engage with end-users, e.g., responding to questions about the individual mathematics homework exercises.

For the requested Bot Builder sample, technical topics of interest include best practices for coordination between workflow engines and dialog systems and for the processing of bidirectional asynchronous messages and events between them. These asynchronous messages or events should be capable of being processed by dialog systems both during and between interactions with end-users. As envisioned, (1) asynchronous events could be raised by workflows, e.g., a simplest case involving a timer, and sent to dialog systems, and (2) asynchronous events could be raised by dialog systems and sent to workflow engines.

Thank you for considering these topics for a new Bot Builder sample.

**Comment by LeeParrishMSFT** [comment]
Thanks for the feedback, we will take this into consideration for future samples. 

@clearab tagging you here for consideration for future sample efforts.

**Comment by AdamSobieski** [comment]
@LeeParrishMSFT, many thanks.

A generalization for the sought sample is that some other asynchronous end-user activity is desired to be coordinated with a dialog system. End-users might be completing mathematics exercises or shopping for products and their asynchronous activities and progressions through them are desired to be coordinated with dialog systems so that the dialog systems can better assist the end-users.

In a mathematics education example, learners progress through sequences of modular exercises (each of which could be considered to be either a dynamically-loaded workflow or a dynamically-loaded software component from a .NET assembly) and the dialog systems are desired to have shared contexts with the asynchronous activities. In this way, dialog systems could better answer contextual questions to assist learners.

In a shopping example, customers might select products via Web-based user interfaces and then ask coordinated dialog systems questions about them. Coordination between asynchronous shopping user experiences and dialog systems can enable better understanding of natural-language questions which make use of shared contexts. Dialog systems could, for instance, utilize shared contexts to resolve exophoric references – such as "this" or "that" – which pertain to customers' user-interface selections.

In conclusion, best practices with respect to the coordination between other asynchronous end-user activities and Bot Framework dialog systems is the gist of the sought sample.

**Comment by vansh-afk** [comment]
> Thanks for the feedback, we will take this into consideration for future samples. 
> 
> @clearab tagging you here for consideration for future sample efforts.

Hi I am new in these field canu you please guide me
